pioneer

```diff
@@ -2,12 +2,20 @@
 # Executable wrapper script
 # Ensures the program is up-to-date, then hands over
 
-PROGRAM_NAME=$(basename $0)
+set -e
+cd "$(dirname "$0")"
+PROGRAM_NAME="$(basename "$0")"
 
-EXEC_BIN=src/$PROGRAM_NAME
-START_DIR=$(pwd)
-WORK_DIR=$(dirname $0)
+# Create build directory
+[ -d build ] || make build
+cd build
 
-cd $WORK_DIR
-make && exec $EXEC_BIN $*
-cd $START_DIR
+# Generate build files
+[ -f Makefile ] || cmake ..
```

Comment:
You can run "cmake .." unconditionally.

Reply:
Doing so regenerates `buildopts.h`, which, in turn, results in rebuilding some files, relinking Pioneer, and re-running the model compiler. The Makefile automatically re-runs CMake whenever necessary.

Reply:
OTOH, my previous observation seems to be related to me switching commits before rerunning the script. It might be still better to print a message (like `warn: version may be stale; re-run cmake manually if you need it up-to-date`) instead of re-running CMake to eliminate those compile times if one does not care about Pioneer displaying the right commit.